src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -133,8 +155,29 @@ export function useRefetchQuoteCallback() {
           console.debug('[useRefetchPriceCallback] Canceled get quote price for', params)
           return
         }
+
         const [price, fee] = data as QuoteResult
 
+        // Promise.allSettled (L68) does NOT throw whole promise on first reject
+        // we don't want it to, we are waiting for FEE and PRICE, if only one fails, why throw both away?
+        // e.g FEE comes in OK, but Fee > Price, we throw an FEE_EXCEEDS_PRICE error but keep fee promise resolved value
+        // to save into state and use
+        quoteData = {
+          ...params.quoteParams,
+          fee: getPromiseFulfilledValue(fee, undefined),
+          price: getPromiseFulfilledValue(price, undefined),
+          lastCheck: Date.now()
+        }
+
+        // check the PromiseSettledValue/PromiseRejectedResult
+        // returned by Promise.allSettled
```

Comment:
Can we try to use a less technical and more descriptive language for the comment?

Reply:
noted in https://github.com/gnosis/cowswap/issues/751